numba/jitclass/_box.c

```diff
@@ -0,0 +1,186 @@
+
+
+static PyTypeObject BoxType = {
+#if (PY_MAJOR_VERSION < 3)
+    PyObject_HEAD_INIT(NULL)
+    0,                         /*ob_size*/
+#else
+    PyVarObject_HEAD_INIT(NULL, 0)
+#endif
+    "_box.Box",                /*tp_name*/
+    sizeof(BoxObject),         /*tp_basicsize*/
+    0,                         /*tp_itemsize*/
+    (destructor)box_dealloc,   /*tp_dealloc*/
+    0,                         /*tp_print*/
+    0,                         /*tp_getattr*/
+    0,                         /*tp_setattr*/
+    0,                         /*tp_compare*/
+    0,                         /*tp_repr*/
+    0,                         /*tp_as_number*/
+    0,                         /*tp_as_sequence*/
+    0,                         /*tp_as_mapping*/
+    0,                         /*tp_hash */
+    0,                         /*tp_call*/
+    0,                         /*tp_str*/
+    0,                         /*tp_getattro*/
+    0,                         /*tp_setattro*/
+    0,                         /*tp_as_buffer*/
+    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE, /*tp_flags*/
+    Box_doc,                   /* tp_doc */
+    0,                         /* tp_traverse */
+    0,                         /* tp_clear */
+    0,                         /* tp_richcompare */
+    0,                         /* tp_weaklistoffset */
+    0,                         /* tp_iter */
+    0,                         /* tp_iternext */
+    0,                         /* tp_methods */
+    0,                         /* tp_members */
+    0,                         /* tp_getset */
+    0,                         /* tp_base */
+    0,                         /* tp_dict */
+    0,                         /* tp_descr_get */
+    0,                         /* tp_descr_set */
+    0,                         /* tp_dictoffset */
+    (initproc)Box_init,        /* tp_init */
+    0,                         /* tp_alloc */
+    PyType_GenericNew,         /* tp_new */
+};
+
+
+/* Import MemInfo_Release from numba.runtime._nrt_python once for use in
+ * Box_dealloc.
+ */
+static
+void* import_meminfo_release() {
+    PyObject *nrtmod = NULL;
+    PyObject *helperdct = NULL;
+    PyObject *mi_rel_fn = NULL;;
+    void *fnptr = NULL;;
+    /* from numba.runtime import _nrt_python */
+    nrtmod = PyImport_ImportModule("numba.runtime._nrt_python");
+    if (!nrtmod) goto cleanup;
+    /* helperdct = _nrt_python.c_helpers */
+    helperdct = PyObject_GetAttrString(nrtmod, "c_helpers");
+    if (!helperdct) goto cleanup;
+    /* helperdct['MemInfo_release'] */
+    mi_rel_fn = PyDict_GetItemString(helperdct, "MemInfo_release");
+    if (!mi_rel_fn) goto cleanup;
+    fnptr = PyLong_AsVoidPtr(mi_rel_fn);
+
+cleanup:
+    Py_XDECREF(nrtmod);
+    Py_XDECREF(helperdct);
+    return fnptr;
+}
+
+/* Debug utils.
+ * Get internal dataptr field from Box.
+ */
+static
+PyObject* box_get_dataptr(PyObject *self, PyObject *args) {
+    BoxObject *box;
+    /* no type checking */
```

Comment:
Well, type safety is useful when debugging ;-) By the way, the `O!` format should make it almost trivial.